modules/localgov_services_sublanding/tests/src/Functional/UnpublishedLinkNodeReference.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\tests\localgov_services_sublanding\Functional;
+
+use Drupal\Tests\BrowserTestBase;
+use Drupal\node\NodeInterface;
+use Drupal\node\Entity\Node;
+use Drupal\Component\Utility\Random;
+
+/**
+ * Functional tests for the LinkNodeReference field formatter.
+ *
+ * Tests that:
+ * - Unpublished child nodes are not listed by the parent page to anonymous
+ *   users.
+ * - When an unpublished node is added as a child page and later published, it
+ *   should be listed by the parent page for anonymous users.
+ * - Unpublished child nodes are clearly highlighted to editors.
+ *
+ * @group localgov_services_sublanding
+ */
+class UnpublishedLinkNodeReference extends BrowserTestBase {
+
+  /**
+   * Setup Service sub page and an editor user capable of editing it.
+   *
+   * Create an unpublished Service page and set it as a child page of a new
+   * Service sub page.  The LinkNodeReference field formatter of the parent page
+   * lists this child page as a teaser.
+   */
+  public function setUp() {
+
+    parent::setUp();
+
+    $this->editorUser = $this->drupalCreateUser([
+      'administer content types',
+      'administer nodes',
+      'bypass node access',
+    ]);
+    $this->drupalLogin($this->editorUser);
+
+    $service_sub_page = $this->drupalCreateNode([
+      'type' => 'localgov_services_sublanding',
+      'title' => self::SERVICE_SUB_PAGE_TITLE,
+      'status' => NodeInterface::PUBLISHED,
+      'body' => [
+        [
+          'summary' => 'No summary',
+          'value' => (new Random)->paragraphs(3),
+          'format' => filter_default_format(),
+        ],
+      ],
+    ]);
+    $this->serviceSubPageLastNid = $service_sub_page->id();
+
+    // Create a Service page which will be used as the child page.
+    $child_page = $this->drupalCreateNode([
+      'title'  => self::CHILD_PAGE_TITLE,
+      'status' => NodeInterface::NOT_PUBLISHED,
+      'type'   => 'localgov_services_page',
+      'localgov_services_parent' => [
+        [
+          'target_id' => $this->serviceSubPageLastNid,
+        ],
+      ],
+    ]);
+    $this->childPageLastNid = $child_page->id();
+
+    // Add child page to Service sub page.
+    $this->drupalGet("node/{$this->serviceSubPageLastNid}/edit");
+    $this->submitForm([], 'Add Topic list builder');
+
+    $this->getSession()->getPage()->fillField('field_topics[0][subform][topic_list_links][0][uri]', self::CHILD_PAGE_TITLE . " ($this->childPageLastNid)");
+    $this->submitForm([], 'Save');
+    $this->assertSession()->pageTextNotContains(self::FORM_ERROR_MSG);
+
+    $this->drupalLogout();
+
+    // Create a cached copy of the Service sub page.
+    $this->drupalGet("node/{$this->serviceSubPageLastNid}");
+  }
+
+  /**
+   * Service sub pages should not list **unpublished** child pages.
+   */
+  public function testDoNotListUnpublishedChildPages() {
+
+    $this->drupalGet("node/{$this->serviceSubPageLastNid}");
```

Comment:
Would it not make sense to do some of these tests in one method? 
I'm thinking particularly looking at the page _logged in_ and then this _anonymous_ one. This would also then check that the cache tags are correct and the anonymous user doesn't get to see what the logged in user saw?

Reply:
I have refactored a bit following your comment.

>  anonymous user doesn't get to see what the logged in user saw?

Both of them should see the same thing once the child page is published.